### PR TITLE
added http range support for getStream and added getRange function

### PR DESCRIPTION
**Pull request by wiserweb**
Implemention as per Swift documention for range request using http range headers.

### CODE REVIEW

File: lib/settings.js [hunk 1]
```diff
@@ -50,13 +50,15 @@ const OBJECT_SETTINGS = Object.freeze({
     contentEncoding:    'Content-Encoding',
     deleteAfter:        'X-Delete-After',
     contentType:        'Content-Type',
-    detectContentType:  'X-Detect-Content-Type'
+    detectContentType:  'X-Detect-Content-Type',
+    'range':             'range',
```
Comment (totherik) [inline lib/settings.js]: Please remove the single-quotes around the key to match the other keys.

File: lib/swifft_object.js [hunk 2]
```diff
@@ -42,8 +41,18 @@ export default class SwiftObject {
     }
 
 
-    getStream(callback) {
-        this.invoke({ stream: true }, (err, res, _) => {
+    getStream(range,callback) {
```
Comment (totherik) [inline lib/swifft_object.js]: Please add a space between arguments.

File: lib/swifft_object.js [hunk 3]
```diff
@@ -53,6 +62,24 @@ export default class SwiftObject {
         });
     }
 
+    getRange(range,callback) {
```
Comment (totherik) [inline lib/swifft_object.js]: Please add a space between arguments.

**Comment by totherik** [comment]
This looks great, @wiserweb, just made a couple formatting comments inline, but thank you!

Prior to merge, could you please add/update appropriate tests for the updated `getStream` signature and the new `getRange` API?

**Comment by wiserweb** [comment]
I have it on a to do list to add an example in readme for getRange, getSteam and add tests.

Will try to push another commit this week.

Thanks for fixing the formatting.

**Comment by wiserweb** [comment]
Unable to test the getRange on this system. it should work... can you please test both getRange and getStream to make sure they pass? Perhaps these two methods will need some basic documentation in readme.

File: lib/swifft_object.js [hunk 4]
```diff
@@ -53,6 +62,24 @@ export default class SwiftObject {
         });
     }
 
+    getRange(range, callback) {
+
+        let opts = {
+            method: 'GET',
+            //payload: content,
+            headers: ObjectSettings.toHeaders(range)
```
Comment (totherik) [inline lib/swifft_object.js:70]: Even though this argument is named `range` it appears that it expects an [explicitly structured object](https://github.com/totherik/swifft/pull/2/files#diff-c4912838eca48e56a1000a78da9532aaR69) with `range` as a property (thus the argument itself isn't "range").

Perhaps, this method should accept a string value for `range` and wrap it appropriately:

``` js
let opts = {
    method: 'GET',
    headers: ObjectSettings.toHeaders({ range })
};

```

**Comment by totherik** [comment]
just the one comment. I'll work on running tests today. thanks, again!

**Comment by wiserweb** [comment]
I'm in need to sharpen my tape skills.

Below is a link to a repo with a working sample of the getRange code
that might be of assistance.

https://github.com/wiserweb/swifft-example-getrange

**Comment by totherik** [comment]
No worries. I think things look ok on that front, just trying to nail down the naming so the API is clear. the current `getRange` API has an argument of `range` but when reading the code, the first argument is actually a "settings" object. Should this API accept various settings as built, knowing the settings object could be missing a "range" property? Or should it just accept an explicit "range" string value and create a settings object with only that property?

It doesn't seem like other settings apply in this case, so only allowing `range` _appears_ to make sense. In that case the signature could be `getRange(String, Function)` instead of `getRange(Object, Function)`

**Comment by wiserweb** [comment]
> > Should this API accept various settings as
> > built, knowing the settings object could be missing a "range" property?

Yes.

Implicitly the getRange method could get the entire object.
i.e. |[range] bytes=0- |

So even if it's omitted it would have a default range (the entire length
of the object) so the method name remains meaningful.

In other cases Range would be explicitly supplied, which implicitly
overrides the default range of |[range] bytes=0- |.

http://developer.openstack.org/api-ref-objectstorage-v1.html

The following forms of the header specify the following ranges of data:

```
Range: bytes=-5. The last five bytes.

Range: bytes=10-15. The five bytes of data after a 10-byte offset.

Range: bytes=10-15,-5. A multi-part response that contains the last
```

five bytes and the five bytes of data after a 10-byte offset. The
Content-Type of the response is then multipart/byteranges.

```
Range: bytes=4-6. Bytes 4 to 6 inclusive.

Range: bytes=2-2. Byte 2, the third byte of the data.

Range: bytes=6-. Byte 6 and after.

Range: bytes=1-3,2-5. A multi-part response that contains bytes 1 to
```

3 inclusive, and bytes 2 to 5 inclusive. The Content-Type of the
response is then multipart/byteranges.

**Comment by totherik** [comment]
Sorry, I trust you on that part, I was more interested in discussing the `getRange` API signature in the code...
- should it just accept an explicit "range" string value and create a settings object with only that property?
- the signature could be `getRange(String, Function)` (the string being the header values you specify above) instead of `getRange(Object, Function)`

**Comment by wiserweb** [comment]
I think String works well but I'm not opposed to using Object. I trust your judgement on this. I suppose it can always be refactored later if there's a need for a different signature.

**Pull request merged**